### PR TITLE
Support FontVariation.Settings in components.resources

### PR DESCRIPTION
**Pull request by adamglin0**
Support Support FontVariation.Settings in components.resources

Fixes [[link to the issue](https://youtrack.jetbrains.com/issue/CMP-7088/Support-FontVariation.Settings-in-compose.component.resource)]

## Testing

Should be manually tested.

### Run on desktop via Gradle:
`./gradlew :resources:demo:desktopApp:run`

### Run JS in browser with WebAssembly Skia via Gradle:
`./gradlew :resources:demo:shared:jsBrowserDevelopmentRun`

This should be tested by QA

## Release Notes

- Support FontVariation.Settings in components.resources

Subsections:
- Resources

### Section - Subsection

- Support FontVariation.Settings in components.resources

### CODE REVIEW

File: components/resources/library/src/commonMain/kotlin/org/jetbrains/compose/resources/FontResources.kt [hunk 1]
```diff
@@ -35,6 +35,15 @@ expect fun Font(
     style: FontStyle = FontStyle.Normal
 ): Font
 
+
+@Composable
+expect fun Font(
```
Comment (terrakok) [inline components/resources/library/src/commonMain/kotlin/org/jetbrains/compose/resources/FontResources.kt:40]: 1) for any public API we need a documantation
2) do we need a function with all parameters? (variation and standard)

File: components/resources/library/src/androidMain/kotlin/org/jetbrains/compose/resources/FontResources.android.kt [hunk 2]
```diff
@@ -6,9 +6,14 @@ import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.text.font.*
 
 @Composable
-actual fun Font(resource: FontResource, weight: FontWeight, style: FontStyle): Font {
```
[the remark this answers is not in the document]
Comment (adamglin0) [reply components/resources/library/src/androidMain/kotlin/org/jetbrains/compose/resources/FontResources.android.kt:9]: Adding parameters with default values can also lead to binary incompatibility🤔？

[the remark this answers is not in the document]
Comment (MatkovIvan) [reply components/resources/library/src/androidMain/kotlin/org/jetbrains/compose/resources/FontResources.android.kt:9]: yep. It's source-comparable but not binary-compatible

**Comment by terrakok** [comment]
https://github.com/JetBrains/compose-multiplatform/pull/5183

**Pull request closed without merging**